iceoryx_utils/include/iceoryx_utils/internal/cxx/string_internal.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_UTILS_CXX_STRING_INTERNAL_HPP
+#define IOX_UTILS_CXX_STRING_INTERNAL_HPP
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+namespace iox
+{
+namespace cxx
+{
+template <uint64_t>
+class string;
+
+namespace internal
+{
+template <uint64_t N>
+using charTemp = char[N];
+
+/// @brief struct to get capacity of fixed string/string literal
+/// @note capa is a dummy value for any type other than cxx::string and char
+template <typename T>
+struct GetCapa
+{
+    static constexpr uint64_t capa = 0;
+};
+
+template <uint64_t N>
+struct GetCapa<string<N>>
+{
+    static constexpr uint64_t capa = N;
+};
+
+template <uint64_t N>
+struct GetCapa<char[N]>
+{
+    static constexpr uint64_t capa = N - 1;
+};
+
+/// @brief struct to get size of fixed string/string literal/std::string
+template <typename T>
+struct GetSize;
+
+template <uint64_t N>
+struct GetSize<string<N>>
+{
+    static uint64_t call(const string<N>& data)
+    {
+        return data.size();
+    }
+};
+
+template <uint64_t N>
+struct GetSize<char[N]>
+{
+    static uint64_t call(const charTemp<N>&)
+    {
+        return N - 1u;
```

Comment:
This is a good point! But I think the usage is alright since this struct is only used to get the maximum size of a result which is produced by a certain method. And in your case you would produce a `cxx::string` of capacity `10` but only with `3` elements inside. 
Here is also the question, what is the actual size of a string literal like `char x[10] = "123";` Is it 3 or 10 or would you differentiate and say it has the capacity of 10 and the size of 3?!

Reply:
I would differentiate, capacity 10 and size 3, since this is also what strlen returns. How about calling this function `GetMaxSize` or `GetCapacity`?

Reply:
We should differentiate between capacity and size. We already have `GetCapa` for the capacity, for `GetSize` I will use `strnlen` to return the correct size.